app/assets/javascripts/random_key.js

```diff
@@ -0,0 +1,19 @@
+function cdxGenRandomKey(items) {
+  var keys;
+  var key;
+
+  if (items) {
+    keys = items.map(function (item) { return item.key });
+  } else {
+    keys = [];
+  }
```

Comment:
This would doit in one line but I'm not sure if it's preferred. 

```
(items||[]).map(function (item) { return item.key });
```

Reply:
Thanks, this is much better!